spring-cloud-task-samples/single-step-batch-job/src/test/java/io/spring/BatchJobApplicationTests.java

```
@@ -0,0 +1,199 @@
+	public void testJdbcReaderJdbcWriter() throws Exception {
+		getSpringApplication().run(SingleStepBatchJobApplication.class,
+			"--spring.profiles.active=jdbcreader,jdbcwriter",
+			"--spring.datasource.username=" + DATASOURCE_USER_NAME,
+			"--spring.datasource.url=" + DATASOURCE_URL,
+			"--spring.datasource.driver-class-name=" + DATASOURCE_DRIVER_CLASS_NAME,
+			"--spring.datasource.password=" + DATASOURCE_USER_PASSWORD);
+		validateDBResult();
+	}
+
+	@Test
+	public void testJdbcReaderFlatfileWriter() throws Exception {
+		getSpringApplication().run(SingleStepBatchJobApplication.class,
+			"--spring.profiles.active=jdbcreader,ffwriter",
+			"--spring.datasource.username=" + DATASOURCE_USER_NAME,
+			"--spring.datasource.url=" + DATASOURCE_URL,
+			"--spring.datasource.driver-class-name=" + DATASOURCE_DRIVER_CLASS_NAME,
+			"--spring.datasource.password=" + DATASOURCE_USER_PASSWORD);
+		validateFileResult();
+	}
+
+	@Test
+	public void testFileReaderFileWriter() throws Exception {
+		getSpringApplication().run(SingleStepBatchJobApplication.class,
+			"--spring.profiles.active=ffreader,ffwriter");
+		validateFileResult();
+	}
+
+	public Server initH2TCPServer() {
+		Server server = null;
+		try {
+			if (defaultServer == null) {
+				server = Server.createTcpServer("-ifNotExists", "-tcp",
+					"-tcpAllowOthers", "-tcpPort", String.valueOf(randomPort))
+					.start();
+				defaultServer = server;
+				DriverManagerDataSource dataSource = new DriverManagerDataSource();
+				dataSource.setDriverClassName(DATASOURCE_DRIVER_CLASS_NAME);
+				dataSource.setUrl(DATASOURCE_URL);
+				dataSource.setUsername(DATASOURCE_USER_NAME);
+				dataSource.setPassword(DATASOURCE_USER_PASSWORD);
+				ClassPathResource setupResource = new ClassPathResource(
+					"schema-h2.sql");
+				ResourceDatabasePopulator resourceDatabasePopulator = new ResourceDatabasePopulator(
+					setupResource);
+				resourceDatabasePopulator.execute(dataSource);
+			}
+		}
+		catch (SQLException e) {
```

I'd just throw the exception in a test...